src/backend/api/Fusion.Resources.Api/Notifications/InternalRequestsNotificationHandler.cs

```diff
@@ -0,0 +1,225 @@
+
+            return new NotificationRequestData(notificationType, internalRequest, orgPosition, orgPositionInstance);
+        }
+        private async Task<QueryResourceAllocationRequest?> GetInternalRequestAsync(Guid requestId)
+        {
+            var query = new GetResourceAllocationRequestItem(requestId);
+            var request = await mediator.Send(query);
+
+            return request;
+        }
+        private async Task<IEnumerable<Guid>> GenerateRecipientsAsync(NotificationRequestData data)
+        {
+            var recipients = new List<Guid>();
+
+            if (data.NotifyCreator)
+                recipients.Add(data.AllocationRequest.CreatedBy.AzureUniqueId);
+
+            if (data.NotifyResourceOwner && data.Instance.AssignedPerson?.AzureUniqueId != null)
+            {
+                var ro = await mediator.Send(new GetResourceOwner(data.Instance.AssignedPerson.AzureUniqueId.Value));
+                if (ro?.IsResourceOwner == true && ro.AzureUniqueId.HasValue)
+                    recipients.Add(ro.AzureUniqueId.Value);
+            }
+
+            if (data.NotifyTaskOwner && data.Instance.TaskOwnerIds != null)
+                recipients.AddRange(data.Instance.TaskOwnerIds);
+
+            return recipients.Distinct();// A person may be a creator and/or resource owner and/or task owner.
+        }
+
+        private class NotificationRequestData
+        {
+            public NotificationRequestData(Type notificationType, QueryResourceAllocationRequest allocationRequest,
+                ApiPositionV2 position, ApiPositionInstanceV2 instance)
+            {
+                AllocationRequest = allocationRequest;
+                Position = position;
+                Instance = instance;
+
+                DecideWhoShouldBeNotified(notificationType, allocationRequest);
+
+                PortalUrl = notificationType.Name switch
+                {
+                    nameof(ResourceAllocationRequest.RequestInitialized) =>
+                        $"/apps/org-admin/{Position.Project.ProjectId}/timeline?instanceId={Instance.Id}&positionId={Position.Id}",
+                    _ => $"/apps/org-admin/{Position.Project.ProjectId}"
+                };
+            }
+
+            private void DecideWhoShouldBeNotified(Type notificationType, QueryResourceAllocationRequest allocationRequest)
+            {
+
+                IsAllocationRequest = allocationRequest.Type == InternalRequestType.Allocation;
+                IsChangeRequest = allocationRequest.Type == InternalRequestType.ResourceOwnerChange;
+
+                var isDirect = allocationRequest.SubType == AllocationDirectWorkflowV1.SUBTYPE;
+                var isJointVenture = allocationRequest.SubType == AllocationJointVentureWorkflowV1.SUBTYPE;
+                var isNormal = allocationRequest.SubType == AllocationNormalWorkflowV1.SUBTYPE;
+
+
+
+                switch (notificationType.Name)
```

Comment:
I personally think this logic would be cleaner if in the future it's refactored into some kind of configuration data. I.e. 
```
class RequestNotificationConfig {
    RequestType,
    RequestSubType,
    NotifyTaskOwner,
    NotifyCreator,
    NotifyResourceOwner
    ...
}
```
and then just look it up in a static dictionary/table based on type and subtype